slsa/provenance.go

```diff
@@ -15,24 +15,33 @@
 package slsa
 
 import (
+	"context"
 	"fmt"
+	"regexp"
 
 	intoto "github.com/in-toto/in-toto-golang/in_toto"
 	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
-
 	"github.com/slsa-framework/slsa-github-generator/github"
 )
 
 const (
 	// GithubHostedActionsBuilderID is the builder ID for Github hosted actions.
 	GithubHostedActionsBuilderID = "https://github.com/Attestations/GitHubHostedActions@v1"
-	audience                     = "slsa-framework"
+)
+
+var (
+	githubComReplace = regexp.MustCompile(`^(https?://)?github\.com/?`)
 )
 
 // HostedActionsProvenance generates an in-toto provenance statement in the SLSA
 // v0.2 format for a workflow run on a Github actions hosted runner.
-func HostedActionsProvenance(w WorkflowRun) (*intoto.ProvenanceStatement, error) {
-	t, err := github.RequestOIDCToken(audience)
+func HostedActionsProvenance(ctx context.Context, w WorkflowRun, c *github.OIDCClient) (*intoto.ProvenanceStatement, error) {
```

Comment:
maybe needed to replace https://github.com/slsa-framework/slsa-github-generator-go/blob/main/.github/workflows/builder.yml#L67-L79?

Reply:
IIUC, this workflow job you're linking to is to get the workflow ref so that we can build the right version of the builder and is tracked in #4. It's early in the morning but I'm not understanding how this is related here.

Reply:
ignore my comment, my bad.